Dockerfile

```diff
@@ -22,8 +22,12 @@ WORKDIR /go/src/github.com/cosmos/cosmos-sdk
 # Add source files
 COPY . .
 
+# Dockerfile Cross-Compilation Guide
+# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide
+ARG TARGETOS TARGETARCH
```

Comment:
Curious, where these are passed as argument? Don't we need `build-arch`?

Reply:
my understanding this value is set at building of the image, if a user runs it on a aarch64 machine it wont infer the change unless you rebuild the image, is that right?

Reply:
Yes, after building you only have the output container. And when you run docker build(x), it pulls the build container for the current platform because I mentioned `--platform=$BUILDPLATFORM`.

| build-arch | target-arch | build-container | output-container |
| :-: | :-: | :-: | :-: |
| `amd64` | `arm64` | `amd64` | `arm64` |
| `arm64` | `amd64` | `arm64` | `amd64` |

So trying to build this on different machines may pull different build containers. But the output containers will be consistent with what you pass in `--platform` param. And the copied go binary is consistent with the output container because we used `GOOS` and `GOARCH` accordingly.

Reply:
makes sense, this is some optimisation for builds it seems